src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -53,8 +53,19 @@ public ActionResult execute() throws EntityDoesNotExistException {
         student.comments = Sanitizer.sanitizeTextField(student.comments);
         
         try {
-            student.updateWithExistingRecord(logic.getStudentForEmail(courseId, studentEmail));
-            logic.validateSections(Arrays.asList(student), courseId);
+            StudentAttributes originalStudentAttribute = logic.getStudentForEmail(courseId, studentEmail);
+            student.updateWithExistingRecord(originalStudentAttribute);
+            
+            boolean hasSectionChanged = student.hasSectionChanged(originalStudentAttribute);
+            boolean hasTeamChanged = student.hasSectionChanged(originalStudentAttribute);
+            if(hasSectionChanged && hasTeamChanged){
```

Comment:
Based on the reason given below, we could omit this condition and simply `validateSectionsAndTeams` if a section has changed.